pkg/httphelper/httphelper_test.go

```diff
@@ -0,0 +1,206 @@
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus/testutil"
+)
+
+func TestWriteHeader(t *testing.T) {
+	testcases := []struct {
+		name       string
+		statusCode int
+	}{
+		{
+			"StatusOK",
+			http.StatusOK,
+		},
+		{
+			"StatusNotFound",
+			http.StatusNotFound,
+		},
+	}
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			rr := httptest.NewRecorder()
+			trw := &TraceResponseWriter{ResponseWriter: rr, statusCode: http.StatusOK}
+			trw.WriteHeader(tc.statusCode)
+			if rr.Code != tc.statusCode {
+				t.Errorf("mismatch between expected and actual response headers: expected %s, got %s", http.StatusText(tc.statusCode), http.StatusText(rr.Code))
+			}
+			if trw.statusCode != tc.statusCode {
+				t.Errorf("mismatch between expected and actual TraceResponseWriter headers: expected %s, got %s", http.StatusText(tc.statusCode), http.StatusText(trw.statusCode))
+			}
+		})
+
+	}
+
+}
+
+func TestWrite(t *testing.T) {
+	testcases := []struct {
+		name         string
+		responseBody string
+	}{
+		{
+			"SimpleText",
+			"Simple text to test respone size",
+		},
+	}
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			rr := httptest.NewRecorder()
+			trw := &TraceResponseWriter{ResponseWriter: rr, statusCode: http.StatusOK}
+			resp := []byte(tc.responseBody)
+			_, err := trw.Write(resp)
+			if err != nil {
+				t.Fatalf("failed to write to TraceResponseWriter")
+			}
+			if rr.Body.String() != tc.responseBody {
+				t.Errorf("mismatch between expected and actual response body: expected %s, got %s", tc.responseBody, rr.Body.String())
+			}
+			if trw.size != len(resp) {
+				t.Errorf("mismatch between expected and actual TraceResponseWriter size: expected %d, got %d", len(resp), trw.size)
+			}
+		})
+	}
+}
+
+var metrics = NewMetrics("testnamespace")
+
+var handleMetricsCalls = 0
+var recordErrorCalls = 0
+
+func TestRecordError(t *testing.T) {
+	testcases := []struct {
+		name          string
+		expectedError string
+		expectedCount int
+		expectedOut   string
+	}{
+		{
+			name:          "Simple Error String",
+			expectedError: "sample error message to ensure proper working",
+			expectedOut: `# HELP testnamespace_error_rate number of errors, sorted by label/type
+					   # TYPE testnamespace_error_rate counter
+					   testnamespace_error_rate{error="sample error message to ensure proper working"} %d
+					   `,
+		},
+	}
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			if metrics.ErrorRate == nil {
+				t.Errorf("errorRate not initialized")
+			}
+			metrics.RecordError(tc.expectedError)
+			recordErrorCalls++
+			expectedOut := fmt.Sprintf(tc.expectedOut, recordErrorCalls)
+			if err := testutil.CollectAndCompare(metrics.ErrorRate, strings.NewReader(expectedOut)); err != nil {
+				t.Errorf("unexpected metrics for ErrorRate:\n%s", err)
+			}
+
+		})
+	}
+}
+func wrapperOneByteWriter(t *testing.T) http.HandlerFunc {
```

Comment:
This isn't wrapping anything

Reply:
Why do you say that isnt it wrapping the function that is being returned? Or do you mean that it doesn't need the "wrapper" since the inner function is anonymous?